content/blog/2021-11-17-exp-versioning.md

```diff
@@ -0,0 +1,300 @@
+
+If the experiment instead looks interesting, you can push it along with your
+code so that they stay in sync and are shared with the same people, and it's
+already organized using your existing branch name. You can keep iterating on
+that branch, start a new one if an experiment diverges too much, or merge into
+your main branch to make it your primary model.
+
+# Benefits of DVC
+
+Of course, you could do all this yourself in Git. Since we have established that
+code version control is insufficient for experiments, DVC adds features specific
+to ML needs.
+
+Large data and models aren't easily tracked in Git, so DVC tracks them using
+your own storage, yet they are Git-compatible. For example, the above
+`dvc exp init` command tracks the `models` folder in DVC, making Git ignore it
+yet storing and versioning it so you can back up versions anywhere and check
+them out alongside your experiment code.
+
+Codifying the entire experiment pipeline is a good first step towards
+reproducibility, but it still leaves the user to execute that pipeline, and ML
+pipelines often take to reproduce on the fly. DVC can reproduce the experiment
+with a single command, and it will check for cached inputs and outputs and skip
+recomputing data that's been generated before.
+
+```dvc
+$ dvc exp run
+'data/images.dvc' didn't change, skipping
+Stage 'default' didn't change, skipping
+
+Reproduced experiment(s): exp-44136
+Experiment results have been applied to your workspace.
+
+To promote an experiment to a Git branch run:
+
+        dvc exp branch <exp> <branch>
+```
+
+While Git branching is a flexible way to organize and manage experiments, there
+are often too many experiments to fit any Git branching workflow. DVC tracks
+experiments so you don't need to create commits or branches for each one:
+
+```dvc
+$ dvc exp show
+┏━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━┳━━━━━━━━━┳━━━━━━━━┳━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━┓
+┃ Experiment              ┃ Created      ┃    loss ┃    acc ┃ train.epochs ┃
+model.conv_units ┃
+┡━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━╇━━━━━━━━━╇━━━━━━━━╇━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━┩
+│ workspace               │ -            │ 0.25183 │ 0.9137 │ 10           │ 64
+│
+│ dvc-init                │ Oct 23, 2021 │       - │      - │ 10           │ 16
+│
+│ ├── 9a4ff1c [exp-333c9] │ 10:40 AM     │ 0.25183 │ 0.9137 │ 10           │ 64
+│
+│ ├── 138e6ea [exp-55e90] │ 10:28 AM     │ 0.25784 │ 0.9084 │ 10           │ 32
+│
+│ └── 51b0324 [exp-2b728] │ 10:17 AM     │ 0.25829 │ 0.9058 │ 10           │ 16
+│
+└─────────────────────────┴──────────────┴─────────┴────────┴──────────────┴──────────────────┘
+```
+
+These experiments can be converted into Git branches:
+
+```dvc
+$ dvc exp branch exp-333c9 conv-units-64
+Git branch 'conv-units-64' has been created from experiment 'exp-333c9'.
+To switch to the new branch run:
+
+        git checkout conv-units-64
+
+```
+
+# History of Experiment Tracking
```

Comment:
That's really nice analogy. However, it might be better to focus on functionality and keep it for a next blog post. If you decide to do that it still worth to keep a paragraph to reference the history.